context/src/main/java/io/micronaut/scheduling/executor/ExecutorFactory.java

```diff
@@ -57,7 +60,13 @@ public ExecutorFactory(BeanLocator beanLocator, ThreadFactory threadFactory) {
      */
     @EachBean(ExecutorConfiguration.class)
     protected ThreadFactory eventLoopGroupThreadFactory(ExecutorConfiguration configuration) {
-        return configuration.getName() == null ? threadFactory : new NamedThreadFactory(configuration.getName() + "-executor");
+        if (configuration.isVirtual()) {
```

Comment:
Can `configuration.getName()` return `null` in case of virtual?

Reply:
yea, not sure what to do then, since we can't use the default threadFactory